### PR TITLE
More docstrings for bitcoin methods [ch1214]

### PR DESCRIPTION
**Pull request by trojkat**

### CODE REVIEW

File: clove/network/bitcoin/base.py [hunk 1]
```diff
@@ -104,7 +168,24 @@ def get_nodes(seed) -> list:
...
     def capture_messages(self, expected_message_types: list, timeout: int=20, buf_size: int=1024,
                          ignore_empty: bool=False) -> list:
-
+        '''
+        Method for receiving messages from network nodes.
+
+        Args:
+            expected_message_types (list): list of message types to search for
+            timeout (int): timeout for waiting for messages
+            buf_size (int): buffer size that is going to be used for reviving messages
```
Comment (pafff) [inline clove/network/bitcoin/base.py]: `reciving`

File: clove/network/bitcoin/base.py [hunk 2]
```diff
@@ -313,15 +543,26 @@ def broadcast_transaction(self, raw_transaction: str):
...
         return transaction_address
 
     @auto_switch_params()
-    def send_inventory(self, serialized_transaction) -> msg_getdata:
+    def send_inventory(self, serialized_transaction) -> Optional[msg_getdata]:
+        '''
+        Sends inventory message with given serialized transaction to connected node.
+
+        Returns:
+            msg_getdata, None: get data request of None if something went wrong
```
Comment (pafff) [inline clove/network/bitcoin/base.py]: `or`

File: clove/network/bitcoin/base.py [hunk 3]
```diff
@@ -387,7 +656,27 @@ def atomic_swap(
...
+        '''
+        Creates atomic swap unsigned transaction object.
+
+        Args:
+            sender_address (str): wallet address of the sender
+            recipient_address (str): wallet address of the recipient
+            value (float): amount to swap
+            solvable_utxo (list): optional list of UTXO objects. If None then it will try to find UTXO automatically
+                by using the `get_utxo` method.
+            secret_hash (str): optional secret hash to used in transaction. If None then the new hash will be generated.
```
Comment (pafff) [inline clove/network/bitcoin/base.py]: `to be used`

File: clove/network/bitcoin/base.py [hunk 4]
```diff
@@ -406,16 +695,78 @@ def audit_contract(
...
+
+        Example:
+            >>> from clove.network import BitcoinTestNet
+            >>> network = BitcoinTestNet()
+            >>> wallet = network.get_wallet('cV8FDJu3JhLED2D7q5L7FwLCus69TajYGEnTWEbNqhzzV9WxMBE7')
+            >>> wallet.address
+            'mtUXc6UJTiwb5FdoEJ2hzR8R1yUrcj3hcn'
+
+        Note:
+            If not private_key has been provided then a new wallet will be generated just like via `get_new_wallet()`.
```
Comment (pafff) [inline clove/network/bitcoin/base.py]: `private_key has not been` or `no private_key has been`

File: clove/network/bitcoin/base.py [hunk 5]
```diff
@@ -406,16 +695,78 @@ def audit_contract(
...
+        '''
         return BitcoinWallet(private_key, encrypted_private_key, password)
 
     @classmethod
     def extract_secret(cls, raw_transaction: str=None, scriptsig: str=None) -> str:
+        '''
+        Extracting secret form Alice redeem transaction (first redeem in the Atomic Swao).
```
Comment (pafff) [inline clove/network/bitcoin/base.py]: `from`

File: clove/network/bitcoin/contract.py [hunk 6]
```diff
@@ -141,6 +182,21 @@ def participate(
...
+            symbol (str): network symbol
+            sender_address (str): wallet address of the transaction creator
+            recipient_address (str): wallet address of the transaction recipient
+            value (float): amount to be send
+            utxo (list): list of UTXO objects. In None UTXO will be gathered automatically if needed
+                (for bitcoin-based networks)
+            token_address (str): address of the token contract if we want to use a token
+
+        Returns:
+            BitcoinAtomicSwapTransaction, None: Atomic Swap transaction object or None if something went wrong.o
```
Comment (pafff) [inline clove/network/bitcoin/contract.py]: It won't necessarily be `BitcoinAtomicSwapTransaction`, it depends on given `symbol`

**Review by pafff — CHANGES_REQUESTED** [review]

File: clove/network/bitcoin/contract.py [hunk 7]
```diff
@@ -141,6 +182,22 @@ def participate(
...
+            sender_address (str): wallet address of the transaction creator
+            recipient_address (str): wallet address of the transaction recipient
+            value (float): amount to be send
+            utxo (list): list of UTXO objects. In None UTXO will be gathered automatically if needed
+                (for bitcoin-based networks)
+            token_address (str): address of the token contract if we want to use a token
+
+        Returns:
+            BitcoinAtomicSwapTransaction, EthereumAtomicSwapTransaction, None: Atomic Swap transaction object
+                or None if something went wrong.o
```
Comment (pafff) [inline clove/network/bitcoin/contract.py]: Remove `o` from the end

**Review by pafff — APPROVED** [review]

File: clove/network/bitcoin/base.py [hunk 8]
```diff
@@ -48,19 +48,36 @@
...
+    '''Message prefix'''
     base58_prefixes = {}
+    '''Dictionary of different prefixes'''
     bitcoin_based = True
+    '''Flag for bitcoin-based networks'''
 
     @classmethod
     def switch_params(cls):
+        '''
+        Method for chaning network parameteres.
```
Comment (jaworows) [inline clove/network/bitcoin/base.py]: changing, parameters

File: clove/network/bitcoin/base.py [hunk 9]
```diff
@@ -104,7 +168,24 @@ def get_nodes(seed) -> list:
...
     def capture_messages(self, expected_message_types: list, timeout: int=20, buf_size: int=1024,
                          ignore_empty: bool=False) -> list:
-
+        '''
+        Method for receiving messages from network nodes.
+
+        Args:
+            expected_message_types (list): list of message types to search for
+            timeout (int): timeout for waiting for messages
+            buf_size (int): buffer size that is going to be used for reciving messages
```
Comment (jaworows) [inline clove/network/bitcoin/base.py]: receiving

**Review by jaworows — APPROVED** [review]

**Pull request merged**